Common/src/CConfig.cpp

```diff
@@ -6008,6 +6009,35 @@ void CConfig::SetMarkers(SU2_COMPONENT val_software) {
     Marker_CfgFile_MixingPlaneInterface[iMarker_CfgFile] = indexMarker;
   }
 
+  /*--- Once we have identified the MixingPlane and Turbomachinery markers
+   *    we next need to determine what type of interface between zones is
+   *    used. It is convenient to do this here as it tidies up the interface
+   *    preproccesing in CDriver ---*/
+  nTurboInterfaces = (nMarker_Turbomachinery -  1)*2; //Two markers per zone
+  Kind_TurboInterface.resize(nTurboInterfaces);
+  for (iMarker_CfgFile = 0; iMarker_CfgFile < nMarker_CfgFile; iMarker_CfgFile++) {
+    /*--- Loop over all interface markers ---*/
+    for (iMarker_ZoneInterface = 0; iMarker_ZoneInterface < nMarker_ZoneInterface; iMarker_ZoneInterface++) {
+      /*--- Identify mixing plane markers ---*/
+      if (Marker_MixingPlaneInterface != nullptr){ // Necessary in cases where no mixing plane interfaces are defined
+        if (Marker_CfgFile_TagBound[iMarker_CfgFile] == Marker_MixingPlaneInterface[iMarker_ZoneInterface]) {
+          /*--- Find which list position this marker is in turbomachinery markers ---*/
+          const auto* target = std::find(&Marker_Turbomachinery[0], &Marker_Turbomachinery[nMarker_Turbomachinery*2-1], Marker_CfgFile_TagBound[iMarker_CfgFile]);
+          const auto target_index = target - Marker_Turbomachinery;
+          /*--- Assign the correct interface ---*/
+          SetKind_TurboInterface(target_index - 1, TURBO_INTERFACE_KIND::MIXING_PLANE);
+        }
+      }
+      if (Marker_Fluid_InterfaceBound != nullptr){ // Necessary in cases where no fluid interfaces are defined
+        if (Marker_CfgFile_TagBound[iMarker_CfgFile] == Marker_Fluid_InterfaceBound[iMarker_ZoneInterface]) {
+          const auto* target = std::find(&Marker_Turbomachinery[0], &Marker_Turbomachinery[nMarker_Turbomachinery*2-1], Marker_CfgFile_TagBound[iMarker_CfgFile]);
+          const auto target_index = target - Marker_Turbomachinery;
+          SetKind_TurboInterface(target_index-1, TURBO_INTERFACE_KIND::FROZEN_ROTOR);
```

Comment:
From investigating the `slinc_steady` test case with a debug build, there is something wrong here.

`Marker_Turbomachinery` happens to be a nullpointer, so that the results of `std::find` are nonsense and you end up with `target_index == -1` and an out-of-bounds access. Also, `nMarkerTurbomachinery == 0`, which breaks both `(nMarker_Turbomachinery - 1)*2` and `nMarker_Turbomachinery*2-1`.

Are there some checks missing? It sounds like this code should not have been executed by the test case in the first place.

Reply:
Added an if to only run this code if turbomachinery markers exists, however I think the issue that is causing the address sanitizer is that a fluid interface is specified in a multizone config file for the test case `slinc_steady`, hence when it loops over the iMarker_ZoneInterface it tries to access 'Marker_Fluid_InterfaceBound[iMarker_ZoneInterface]` for a position that doesn't exist in that zone - hence why it throws the error when parsing zone 1, and not when parsing the whole config at the start of the simulation. I have replaced 'Marker_CfgFile_TagBound[iMarker_CfgFile]' with 'Marker_CfgFile_KindBC[iMarker_CfgFile]' as this may be an issue in the future.

'target_index - 1' exists as this sets the interface in the correct zone when it is read in CDriver by essentially removing the inlet from the list. It was a trade-off between doing something that looks weird here, or reformat the whole interface preprocessing. I guess you could also restrict the 'std::find' but again you would run into something that looks a bit strange, this just seems to me to be the simplest way to do it - open to any suggestions on this, I will clear it up with a comment.

There is a lot of strong assumptions here, mainly because elsewhere in the code MARKER_TURBOMACHINERY is assumed to be the flow path of the machine from inlet to outlet.

Reply:
Well I fixed the address sanitizer; just broke something else in doing so 😄

Reply:
@jblueh Appeared to have fixed these issues, would appreciate your input

Reply:
Maybe you can make some of the underlying assumptions visible by assertions? See a suggestion in another comment of mine. I can't judge whether the change to `Marker_CfgFile_KindBC` is the correct solution here, maybe someone else can comment?